fhir-swagger-generator/src/main/java/com/ibm/fhir/openapi/generator/FHIROpenApiGenerator.java

```diff
@@ -934,20 +1141,33 @@ private static void generateSearchViaPostPathItem(Class<?> modelClass, JsonObjec
     }
 
     private static void generateSearchFormParameters(Class<?> modelClass, JsonObjectBuilder parameters) throws Exception {
-        List<SearchParameter> searchParameters = new ArrayList<SearchParameter>(
-                SearchUtil.getApplicableSearchParameters(modelClass.getSimpleName()));
-        for (SearchParameter searchParameter : searchParameters) {
-            String name = searchParameter.getName().getValue();
-
-            JsonObjectBuilder propertyValues = factory.createObjectBuilder();
-            /**
-             * "<name>": { "type": "string" }
-             */
-            propertyValues.add("type", "string");
-            propertyValues.add("description", searchParameter.getDescription().getValue());
-
-            parameters.add(name, propertyValues);
-        }
+        //
+        // As of OpenAPI 3.0.0, empty form fields are still passed to the REST API, which is not valid
+        // for the FHIR search API.
```

Comment:
very interesting.  do we allow it in lenient mode?
is there any discussion in the OpenAPI community about whether this is considered a tooling bug?

Reply:
I did find this: https://github.com/swagger-api/swagger-ui/issues/5303
It seems to indicate there should be a checkbox on the UI to choose whether parameters are passed on the REST call, but I wasn't sure about how to move to a newer version.
I can try and see what happens in lenient mode.

Reply:
With lenient, any parameters for which empty string is invalid are skipped over, but parameters such has _id, code, etc. which allow an empty string, cause the search to not find anything since there are no matches to those values.

Reply:
If I use https://editor.swagger.io/ there is a "Send empty values" checkbox on the form fields. However, with the openliberty openapi UI the IBM FHIR Server currently uses, the checkbox are not shown. So, for now, I'd prefer to just not show those fields. The user can still pass in the form fields on the UI in the request body.

Reply:
thanks for checking on it.